src/Facebook/Facebook.php

```diff
@@ -327,7 +350,7 @@ public function getRedirectLoginHelper()
    */
   public function getJavaScriptHelper()
   {
-    return new FacebookJavaScriptHelper($this->app);
+    return new FacebookJavaScriptHelper($this->app, $this->getOAuth2Client());
```

Comment:
Ah, I see what you mean. But since the `OAuthClient` methods are part of the public API, that means it would be up to the developer to inject the `FacebookApp` instance with each method. Which is quite inconvenient. Plus, injecting a dependency into every method is a code-smell indicating it should be constructor-injected.

But the way I've got it set up in this PR, the `FacebookApp` instance being injected into the signed request helpers and the instance injected into the `OAuth2Client` could be different and all kinds of bad things could happen once those two out-of-sync dependancies make it into the helper. 

How about all the helpers get a `FacebookApp` and `FacebookClient` on instantiation. Then the helpers create an instance of `OAuth2Client` by passing along those two dependancies. That way they are never our of sync and the developer doesn't have to inject a `FacebookApp` with every method call.

```php
class FacebookRedirectHelper {
     public function __construct(FacebookApp $app, FacebookClient $client) {
            $this->oAuth2Client = new OAuth2Client($app, $client);
            // . . 
     }
}
```

Would that work for you? :)